src/UI/components/map/utils/speciesColorMapper.ts

```diff
@@ -0,0 +1,50 @@
+export function createHue(numColor: any, totalColors: any) {
+  if (totalColors === 0) {
+    totalColors = 1;
+  }
+  return ((numColor * (360 / totalColors)) % 360) / 360;
+}
+
+export function speciesColorMapRGB(
+  arrayofSpeciesObjects: any,
+  speciesName: string
+) {
+  const numSpecies = arrayofSpeciesObjects.length;
+  const speciesColorMapObject = arrayofSpeciesObjects.map(
+    (species: any, index: GLfloat) => ({
+      ...species,
+      color: hslToRgb(createHue(index, numSpecies), 1.0, 0.5),
+    })
+  );
+  return speciesColorMapObject.find((l: any) => l.species === speciesName);
+}
+
+export function hslToRgb(h: any, s: any, l: any) {
```

Comment:
Is there not a library that can do this for us? Looks like a lot of test cases will be needed to test this logic properly.

Reply:
Moved away from this method to using hash instead. If we return to this I will look around for an appropriate library